### PR TITLE
Fixing APPCLOUD-294 Limiting the running container count

### PR DESCRIPTION
**Pull request by kasundsilva**

### CODE REVIEW

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java [hunk 1]
```diff
@@ -286,4 +286,8 @@
 
 	public static final String GET_TENANT_APPLICATION_COUNT = "SELECT COUNT(*) FROM AC_APPLICATION WHERE tenant_id = ? " +
```
Comment (danushkaf) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java]: If you changed the existing method why keeping the current query. I believe there are no usages of this anymore

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java [hunk 2]
```diff
@@ -1630,18 +1630,18 @@ public void deleteDeployment(Connection dbConnection, String versionHashId, int
...
         try {
-            preparedStatement = dbConnection.prepareStatement(SQLQueryConstants.GET_TENANT_APPLICATION_COUNT);
+            preparedStatement = dbConnection.prepareStatement(SQLQueryConstants.GET_TENANT_RUNNING_CONTAINER_COUNT);
             preparedStatement.setInt(1, tenantId);
             preparedStatement.setString(2, cloudType);
             resultSet = preparedStatement.executeQuery();
             dbConnection.commit();
             if (resultSet.next()) {
-                appCount = resultSet.getInt(1);
+                appCount = resultSet.getInt("ACTIVE_CONTAINERS_COUNT");
```
Comment (danushkaf) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: Make this a constant

File: modules/jaggeryapps/appmgt/src/site/blocks/application/block.jag [hunk 3]
```diff
@@ -38,7 +38,7 @@ jagg.block("application", {
         var appTypesProperties = require(parsedObject);
         var appType = appTypesProperties[inputs.appTypeName];
         var applicationName = inputs.applicationName;
-        var applicationCount = applicationModule.getApplicationCount();
+        var applicationCount = applicationModule.getRunningApplicationVersionCount();
```
Comment (danushkaf) [inline modules/jaggeryapps/appmgt/src/site/blocks/application/block.jag]: better to change variable name

File: modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag [hunk 4]
```diff
@@ -50,14 +50,21 @@ jagg.block("home", {
         var dashboardUrl = managerModule.getPropertyValue(DASHBOARD_BASE_URL);
 
         var pageTitle = managerModule.getCloudTypeProperties().pageTitle;
+        var maxAppCount = applicationModule.getMaxInstanceCount();
+        var applicationCount = applicationModule.getRunningApplicationVersionCount();
```
Comment (danushkaf) [inline modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag]: better to change variable name

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag [hunk 5]
```diff
@@ -64,7 +64,7 @@ jagg.template("application", function (inputs, outputs, jagg) {
     const APP_TYPE_EXTENSION = "<%=appType.extension%>";
     var applicationName = "<%=applicationName%>";
     var cloudSpecificApplicationRepresentation = "<%=cloudSpecificApplicationRepresentation%>";
-    var applicationCount = <%=applicationCount%>
+    var applicationCount = <%=applicationCount%>;
```
Comment (danushkaf) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: better to change variable name

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag [hunk 6]
```diff
@@ -37,6 +37,9 @@ jagg.template("home", function(inputs, outputs, jagg) {
     var encodedLabels = encodeURIComponent(stringify(selectedApplicationRevision.tags), "UTF-8");
     var encodedEnvs = encodeURIComponent(stringify(selectedApplicationRevision.runtimeProperties), "UTF-8");
     var defaultAppLaunchURL = selectedApplicationRevision.deploymentURL;
+    var applicationCount = outputs.applicationCount;
```
Comment (danushkaf) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag]: better to change variable name

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java [hunk 7]
```diff
@@ -286,4 +286,8 @@
...
             "AND cloud_id=(select id from AC_CLOUD WHERE name=?)";
+
+    public static final String GET_TENANT_RUNNING_CONTAINER_COUNT = "SELECT COUNT(id) AS ACTIVE_CONTAINERS_COUNT FROM " +
```
Comment (amalkasubasinghe) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java]: I prefer to use running_application_version_count, rather using CONTAINER_COUNT

Comment (danushkaf) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java]: +1

File: modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag [hunk 8]
```diff
@@ -50,14 +50,21 @@ jagg.block("home", {
...
+        var cloudSpecificApplicationRepresentation = jagg.module("manager").getCloudTypeProperties().cloudSpecificApplicationRepresentation;
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag]: use managerModule variable here

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag [hunk 9]
```diff
@@ -398,27 +403,32 @@ jagg.template("home", function(inputs, outputs, jagg) {
...
+                conSpecCpu: conSpecCpu,
+                conSpecMemory: conSpecMemory
+
+            }, function(result) {
+                var versions = jQuery.parseJSON(result);
+                //check if application has more than one version
                 jagg.message({
-                    content: "You can start only one version of an application at a time on a free subscription." +
-                        " Other versions were stopped automatically.",
+                    content: "New version started successfully",
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag]: remove jagg.message

File: modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag (hunk 8 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag]: This is fixed https://github.com/wso2/app-cloud/commit/5deb117c126621e6a2555edaf74f470afa0c271b

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag (hunk 9 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag]: This is been fixed.

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java (hunk 7 above)
Comment (kasundsilva) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java]: ****

Comment (kasundsilva) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java]: https://github.com/wso2/app-cloud/pull/703

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag (hunk 6 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/home/template.jag]: https://github.com/wso2/app-cloud/pull/703

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag (hunk 5 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: https://github.com/wso2/app-cloud/pull/703

File: modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag (hunk 4 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/blocks/home/block.jag]: https://github.com/wso2/app-cloud/pull/703

File: modules/jaggeryapps/appmgt/src/site/blocks/application/block.jag (hunk 3 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/blocks/application/block.jag]: https://github.com/wso2/app-cloud/pull/703

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java (hunk 2 above)
Comment (kasundsilva) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: https://github.com/wso2/app-cloud/pull/703

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java (hunk 1 above)
Comment (kasundsilva) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/SQLQueryConstants.java]: https://github.com/wso2/app-cloud/pull/703

**Pull request merged**